README.md

```diff
@@ -302,6 +335,74 @@ class TritonPythonModel:
         return responses
 ```
 
+
+#### Decoupled mode \[Beta\]
+
+This mode allows user to send multiple responses for a request or
+not send any responses for a request. A model may also send
+responses out-of-order relative to the order that the request batches
+are executed. Such models are called *decoupled* models.
+
+
+In decoupled mode, model must use `InferenceResponseSender` object per
+request to keep creating and sending any number of responses for the
+request. The workflow for in this mode looks like:
+
+* `execute` function receives a batch of pb_utils.InferenceRequest as a
+  length N array.
+
+* Iterate through each pb_utils.InferenceRequest and perform for the following
+  steps for each pb_utils.InferenceRequest object:
+
+  1. Get `InferenceResponseSender` object for the InferenceRequest using
+     InferenceRequest.get_response_sender().
+
+  2. Create and populate pb_utils.InferenceResponse to be sent back.
+
+  3. Use InferenceResponseSender.send() to send the above response. If
+     this is the last request then pass pb_utils.TRITONSERVER_RESPONSE_COMPLETE_FINAL
+     as a flag with InferenceResponseSender.send(). Otherwise continue with
+     Step 1 for sending next request.
+
+* The return value for `execute` function in this mode should be None.
+
+Similar to above, in case one of the inputs has an error, you can use
+the `TritonError` object to set the error message for that specific
+request. After setting errors for an pb_utils.InferenceResponse
+object, use InferenceResponseSender.send() to send response with the
+error back to the user.
+
+##### Special Cases
+
+The decoupled mode is powerful and supports various special cases:
+
+* If the model should not send any response for the request,
+  then call InferenceResponseSender.send() with no response
+  but flag parameter set to pb_utils.TRITONSERVER_RESPONSE_COMPLETE_FINAL.
+
+* The model can also send responses out-of-order in which it received
+  requests.
+
+* The request data and `InferenceResponseSender` object can be passed to
+  a separate thread in the model. This means main caller thread can exit
+  from `execute` function and the model can still continue generating
+  responses as long as it holds `InferenceResponseSender` object.
+
+
+The [decoupled examples](examples/decoupled/README.md) demonstrate
+full power of what can be acheived from decoupled API. Read
+[Decoupled Backends and Models](https://github.com/triton-inference-server/server/blob/main/docs/decoupled_models.md)
+for more details on how to host a decoupled model.
+
+##### Known Issues
+
+The support for decoupled models is still in beta and suffers
+from below known issues:
+
+* The decoupled mode does not support [GPU tensors](#interoperability-and-gpu-support).
```

Comment:
"The decoupled mode doesn't support `FORCE_CPU_ONLY_INPUT_TENSORS` parameter to be turned off. This means that the input tensors will always be in CPU."

On a separate bullet point:

"Currently, the InferenceResponseSender.send method only supports inference_response objects that contain only CPU tensors."